scheduler/src/cook/cached_queries.clj

```diff
@@ -16,3 +19,20 @@
   "Given a job entity, return the user the job runs as."
   [job-ent]
   (caches/lookup-cache-datomic-entity! caches/job-ent->user-cache :job/user job-ent))
+
+(defn instance-uuid->job-uuid-datomic-query
+  "Queries for the job uuid from an instance uuid.
+   Returns nil if the instance uuid doesn't correspond
+   to a job"
+  [db instance-uuid]
+  (->> (d/entity db [:instance/task-id (str instance-uuid)])
+       :job/_instance
+       :job/uuid))
+
+(let [miss-fn
+      (fn [instance-uuid]
+        (instance-uuid->job-uuid-datomic-query (d/db datomic/conn) instance-uuid))]
```

Comment:
Nitpick (feel free to ignore):

I think you could instead use [`partial`](https://clojuredocs.org/clojure.core/partial):

```suggestion
      (partial instance-uuid->job-uuid-datomic-query (d/db datomic/conn))]
```

Reply:
I think I'll stick with defining the function as-is for now because it's a bit more readable to me. We would also only ever call the partial function with exactly 1 additional argument so I don't think it's necessary to use `partial`.

(With that said I'm still new to Clojure, so if it's a highly recommended convention to use partial just let me know.)

Reply:
Sounds good, I think either way is fine